src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -1307,15 +1307,20 @@ class EditorObjectSelectionWindow final : public Window
 
     bool FilterSelected(uint8_t objectFlag)
     {
-        if (_FILTER_SELECTED == _FILTER_NONSELECTED)
+        // Track Manager has no concept of selection filtering, so always return true
+        if (gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER)
+        {
+            return true;
+        }
+        else if (_FILTER_SELECTED == _FILTER_NONSELECTED)
         {
             return true;
         }
-        if (_FILTER_SELECTED && objectFlag & ObjectSelectionFlags::Selected)
+        else if (_FILTER_SELECTED && objectFlag & ObjectSelectionFlags::Selected)
```

Comment:
There is no point using else if we’re returning anyway, so please drop it from lines 1315, 1319 and 1323.

Reply:
Removed